pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/SegmentAssignmentUtils.java

```diff
@@ -87,6 +87,15 @@ private SegmentAssignmentUtils() {
     return instances;
   }
 
+  static List<String> getInstancesForDimTable(InstancePartitions instancePartitions) {
+    Preconditions
+            .checkState(instancePartitions.getNumReplicaGroups() == 1 && instancePartitions.getNumPartitions() == 1,
+                    "Instance partitions: %s should contain 1 replica and 1 partition for a dim table",
```

Comment:
why only 1 replica? I thought it will be on every host, so it shall be as many as the hosts?

Reply:
Removed this way of gathering hosts and now getting hosts from a given tag directly. In this case the replication number as such will be disregarded and by default we will assign to all the segment to all the hosts